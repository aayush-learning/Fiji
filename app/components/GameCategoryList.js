import React, { PureComponent } from 'react'

import { FlatList, View, Text, TouchableOpacity, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'
import GameTitle from './GameTitle'

export default class GameCategoryList extends PureComponent {



  _keyExtractor = (item, index) => item._id

  _renderItem = ({item}) => (
    /*<Text>
      { item.category }
    </Text>*/
      <GameTitle
      title={item}
      onPressItem={this.props.onPressItem}
      />
    
  )

  render() {
    return (
      <View style={styles.GameCategoryListStyle}>
        <FlatList
          showsVerticalScrollIndicator={false}
          data={this.props.games}
          keyExtractor={this._keyExtractor}
          renderItem={this._renderItem}
        />
      </View>
    )
  }
}

GameCategoryList.propTypes = {
  games: PropTypes.array,
  onPressItem: PropTypes.func
}

const styles = StyleSheet.create({
  GameCategoryListStyle:{
    flex:1,
    backgroundColor:'white',
  },
  // CategoryCardViewStyle:{
  //   flex:1, 
  //   height:150, 
  //   //width:330, 
  //   marginTop:'8%',
  //   marginBottom:'8%',
  //   //alignSelf:'center',
  //   marginHorizontal:'4%',
  // },
  // CategoryHeaderViewStyle:{
  //   flex:2 ,
  //   justifyContent:'center', 
  //   alignItems:'center', 
  //   backgroundColor:'#dadbc5', 
  //   borderTopLeftRadius:20, 
  //   borderTopRightRadius:20
  // },
  // CategoryHeaderStyle:{
  //   fontSize:25, 
  //   fontWeight:"bold", 
  //   color:'#19a4f2'
  // },
  // CategoryGamesViewStyle:{
  //   //padding:'4%',
  //   //paddingTop:'7%',
  //   //paddingBottom:'7%',
  //   //paddingBottom:'4%',
  //   backgroundColor:'#eaefef',
  //   //backgroundColor:'red',
  //   //justifyContent:'space-around', 
  //   alignContent:"center", 
  //   borderBottomLeftRadius:20, 
  //   borderBottomRightRadius:20
  // },
  // /*EachGameButtonStyle:{ 
  //   //justifyContent:'center',
  //   //alignItems:'center',
  //   //alignContent:"space-around",
  //   //marginHorizontal:'2%,
  //   backgroundColor:'red',
  //   //borderWidth:4,
  //   //borderRadius:50,
  //   paddingRight:'2%',
  //   paddingLeft:'2%'
  //  },*/
  // /*EachGameViewStyle:{
  //   //borderColor:'black',  
  //   //borderWidth:1,
  //   //justifyContent:'center',
  //   borderRadius:40, 
  //   height:85, 
  //   width:85,
  //   backgroundColor:'black',
  //   //borderWidth:2,
  //   marginTop:'8%',
  //   marginBottom:'8%',
  //   //paddingLeft:"4%"
  // },*/
  // /*TextStyle:{
  //   fontSize:20,
  //   fontWeight:'bold',
  //   color:'black',
  // },*/
  // IconViewStyle:{
  //   flex:1,
  //   justifyContent:"center",
  //   //backgroundColor:'black',
  //   //alignItems:"center",
  //   paddingRight:'3%',
  //   //backgroundColor:'black',
  //   //marginBottom:'5%',
  //   //marginTop:'5%'
  // },
  // IconStyle:{
  //   color:'#19a4f2'
  // }
});


/*
<TouchableOpacity style={styles.EachGameButtonStyle}>
            <Text style={styles.TextStyle}>{item.name}</Text>
          </TouchableOpacity>
*/

/*
<TouchableOpacity style={styles.EachGameButtonStyle}>
  <View style={styles.EachGameViewStyle}/>
</TouchableOpacity>

*/