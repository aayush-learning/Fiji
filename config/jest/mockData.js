import { fetchUsersFailure } from '../../app/redux/users';

const contentJson = require('../seed/content.json')
const usersJson = require('../seed/users.json')
const userAliceJson = require('../seed/user_alice.json')
const storyJson = require('../seed/story.json')

export const titles = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('storytitle:')
})

export const stories = storyJson['docs'].filter(function (doc) {
  return doc._id.startsWith('story:')
})

export const games = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('game:')
})

export const gameThemeData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('gametheme:')
})

export const stickersData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('sticker:caterpillar:')
})

export const stickerPacksData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('sticker-pack:')
})

export const highScoresData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('high-score:')
})

export const story_alice_in_wonderland = storyJson['docs'].filter(function (doc) {
  return doc._id == 'story:alice-in-wonderland'
})[0]

export const users = usersJson['docs']

export const messages = userAliceJson['docs'].filter(function(doc) {
  return doc._id.startsWith('chat:mad-hatter:')
})