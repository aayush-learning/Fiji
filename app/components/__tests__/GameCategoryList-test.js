import 'react-native';
import React from 'react';
import GameCategoryList from '../GameCategoryList'
import { games } from '../../../config/jest/mockData'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <GameCategoryList
      games = { games }
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
