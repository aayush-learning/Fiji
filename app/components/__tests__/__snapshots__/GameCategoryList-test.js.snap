// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`this is user 0 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
